### DISCUSSION TITLE
Enhancement: Copy UI and features of ChatGPT mobile app

### DISCUSSION THREAD

**Discussion by adamerose** [danny-avila/LibreChat#2609] [Feature Requests & Suggestions]
## What features would you like to see added?

I think LibreChat when viewed on mobile should have UX similar to the ChatGPT mobile app.

Here is what each UI looks like: https://imgur.com/34Q8Qad
Here is me interacting with the Android app: https://streamable.com/yfxbm8. 
The iPhone version seems to look and behave the same as Android.

### Buttons

Five buttons along the bottom
  - Take Photo (opens camera app)
  - Attach Image (opens photo gallery app)
  - Attach File
  - Voice-to-text
  - Voice conversation mode 

### Gestures

  - Swipe right to show left side panel, swipe left to go close it and return to chat. (#2340)
  - Same thing for right side panel.
  - When already scrolled to top, swipe down in left side panel to refresh conversation data

### UI Tweaks
  - Hide the side arrows. Instead open panels with swipe and/or corner buttons <img width="547" alt="image" src="https://github.com/danny-avila/LibreChat/assets/10884874/576b45db-c390-475e-aea5-f254c822ca6e">
  - Support smaller screens. The UI overflows in settings and config menus. <img width="757" alt="image" src="https://github.com/danny-avila/LibreChat/assets/10884874/e8dffbe1-3dc2-425b-99a4-525a2cd6a490">
  - Hide footer text

### Voice conversation mode
  - Looks like this: https://youtu.be/Cp8KKUxTFaE?t=50
  - This probably needs a dedicated issue, just mentioning it here since it's another feature specifically available in the ChatGPT mobile app and not web.

### Which components are impacted by your request?

UI

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by danny-avila** [comment]
I agree but it's easier said than done. Even ChatGPT on mobile doesn't look like the ChatGPT app.

**Reply by danny-avila** [reply]
Some improvements will be made, but wouldn't be exactly like the ChatGPT app.

**Reply by adamerose** [reply]
Any thoughts on whether LibreChat should have a native mobile app in the roadmap? I think everything I listed so far is doable in a PWA, but it might be good to list out any desired features for LibreChat that aren't possible.

For example I saw [this article](https://arstechnica.com/gadgets/2024/01/android-users-could-soon-replace-google-assistant-with-chatgpt/) about ChatGPT possibly having Google Assistant integration and the ability to read from your screen. I think being able to do that with LibreChat would be awesome.

There's also [Capacitor](https://capacitorjs.com/) which gives more code reuse than React Native but more native integration than PWA.
